contracts/HubPool.sol

```diff
@@ -214,26 +230,22 @@ contract HubPool is Testable, Lockable, MultiCaller, Ownable {
         bytes32 destinationDistributionRoot
     ) public {
         // The most recent refund proposal must be fully claimed before the next relayer refund bundle is initiated.
-        require(
-            relayerRefundRequests.length == 0 || // If this is the first initiated relay refund.
-                relayerRefundRequests[getNumberOfRelayRefunds()].unclaimedPoolRebalanceLeafs == 0,
-            "Last bundle has unclaimed leafs"
-        );
+        require(refundRequest.unclaimedPoolRebalanceLeafCount == 0, "Last bundle has unclaimed leafs");
 
         uint64 requestExpirationTimestamp = uint64(getCurrentTime() + refundProposalLiveness);
 
-        RelayerRefundRequest storage relayerRefundRequest = relayerRefundRequests.push();
-        relayerRefundRequest.requestExpirationTimestamp = requestExpirationTimestamp;
-        relayerRefundRequest.unclaimedPoolRebalanceLeafs = poolRebalanceLeafCount;
-        relayerRefundRequest.poolRebalanceRoot = poolRebalanceRoot;
-        relayerRefundRequest.destinationDistributionRoot = destinationDistributionRoot;
-        relayerRefundRequest.proposer = msg.sender;
+        delete refundRequest; // Remove the existing information relating to the previous relayer refund request.
+
+        refundRequest.requestExpirationTimestamp = requestExpirationTimestamp;
+        refundRequest.unclaimedPoolRebalanceLeafCount = poolRebalanceLeafCount;
+        refundRequest.poolRebalanceRoot = poolRebalanceRoot;
+        refundRequest.destinationDistributionRoot = destinationDistributionRoot;
+        refundRequest.proposer = msg.sender;
```

Comment:
All things being equal I prefer using key-value syntax to set `refundRequest` here because it means that the developer is less likely to forget to set a parameter. Are there gas savings to setting each param individually and leaving default values as-is for keys like `proposerBondRepaid`?

Reply:
so that would replace the `delete` call plus this assigmeneet with this:

```solidity
// Remove the existing information relating to the previous relayer refund request and set to the new info.
        refundRequest = RefundRequest({
            requestExpirationTimestamp: requestExpirationTimestamp,
            unclaimedPoolRebalanceLeafCount: poolRebalanceLeafCount,
            poolRebalanceRoot: poolRebalanceRoot,
            destinationDistributionRoot: destinationDistributionRoot,
            claimedBitMap: 0,
            proposer: msg.sender,
            proposerBondRepaid: false
        });
```

I tried out the gas costs:
using the old assignment + delete that I had cost: 155863
using the syntax above cost: 156041

I think the reason the original syntax is cheaper is you are deleting storage and then not re-asigining some things, such as the `claimedBitMap` and the `proposerBondRepaid`. for these variables the default (post delete values) are the correct in this context.

As a result, I'm going leave it with the original implementation, unless you have anything else I can try that is syntactically better but still gas efficient.

Reply:
I also think there are some annoying solidity inefficiencies with key value syntax. I think they create the entire struct in mem and then assign it rather than just assigning each value directly to storage. This increased mem usage might be why the gas costs are higher.

Reply:
sounds good to me